_schema.graphql

```diff
@@ -2477,6 +2477,8 @@ type City {
 type CitySponsoredContent {
   introText: String
   artGuideUrl: String
+  featuredShows: [Show]
+  shows(size: Int, sort: PartnerShowSorts, status: EventStatus): [Show]
```

Comment:
So this `featuredShows` vs `shows` business...

In the designs we have these content requirements…

![Screen Shot 2019-03-09 at 6 13 13 PM](https://user-images.githubusercontent.com/140521/54078558-0520ab00-4298-11e9-8dcd-07f77cb9d37d.png)


For the featured shows, seems like there are a couple of approaches.

1. Just use the `shows` field in both places

<img width="908" alt="x2" src="https://user-images.githubusercontent.com/140521/54078566-2da8a500-4298-11e9-9e36-49d1c55504b7.png">


2. Have a dedicated `featuredShows` field (which is the third commit in this PR)

<img width="908" alt="featured shows" src="https://user-images.githubusercontent.com/140521/54078554-fdf99d00-4297-11e9-8006-5a579d7d6376.png">

Seems to me the advantages of approach 2 is that it allows us to fiddle with the definition of what "featured shows" means server-side, without Emission having to worry about those details — and thus not constraining us to Emission release cycles, esp v1 vs v1.1

But that also seems like a slippery slope we could go down for any number of fields, so I'm not sure if it's the right idea. 

Thoughts?

Reply:
@anandaroop I don't quite understand the first example – would we add a(n option) parameter to the `shows` resolver requesting sponsored content? 

You raise a really good point about forward-compatible schemas. I think if there's a decent chance that we'll want to include supplementary info beyond the basic `Show` type, we could wrap the `Show` inside a `FeaturedShow` object (even if it only has a single field for now). In that case, we'd need to use a different field name in `sponsoredContent` than `shows()`). But it really depends on what we think we'll want to do, design-wise, after v1.

Reply:
It's not that I anticipate that we'll be adding more info to these featured show bricks. I think rather I'm anticipating that we'll want to change the criteria for _which_ shows we'll feature in these bricks. And that we might want to do so in a way that newer client versions will just work.

But this is all speculative, so maybe in the interest of keeping things moving I'll get rid of the special `city.sponsoredContent.featuredShows` field, and instead let consumers use the more general `city.sponsoredContent.shows`, as in the screencap for approach # 1 above:

<img width="908" alt="shows x22" src="https://user-images.githubusercontent.com/140521/54078620-39489b80-4299-11e9-9dda-75fc329f10dd.png">

The worse case is that v1 clients will show different shows in those preview bricks. Probably not the end of the world.